[ZSA] Fees for ZSA issuance. (#667)
zcash/zips: zip-0317.rst

specified in [#zip-0313]_, and imposes a non-trivial cost for
potential denial-of-service attacks.

ZSA Fee calculation
-------------------

> In addition to the parameters defined in the `Fee calculation`_ section, the ZSA protocol upgrade defines the following additional parameter:

Reviewer: ```suggestion
In addition to the parameters defined in the `Fee calculation`_ section, the ZSA protocol upgrade defines the following additional parameters:
```

===================================== =============== ===============================================
Parameter                                 Value         Units
===================================== =============== ===============================================
:math:`issuance\_fee`                 :math:`1000000` zatoshis per issuance action (as defined below)
:math:`finalize\_fee`                 :math:`1000000` zatoshis per issuance action (as defined below)
===================================== =============== ===============================================

Wallets implementing this specification SHOULD use a conventional fee

> calculated in zatoshis per the following formula:

Reviewer: Does the `zsa_conventional_fee` replace the `conventional_fee` after the ZSA upgrade? Or is it only used in certain circumstances?

This isn't clear, can we specify it here?

Author: Updated.

Reviewer: Thanks!

.. math::

   \begin{array}{rcl}
     zsa\_logical\_actions  &=& logical\_actions \;- \\

>                        & & nActionsOrchard \;+ \\

Reviewer: Why does the formula subtract the number of Orchard actions (nActionsOrchard)?

This means that the non-ZSA Orchard actions are not charging any fee, which is a denial of service risk.

Reviewer: It might be clearer to keep`nActionsOrchard` in the total (don't subtract it), and add a note that it includes both wrapped and native assets.

Author: Updated

of inputs/outputs to be non-proportional above the :math:`grace\_actions`
threshold. This is no longer expressible with the formula specified above.)

ZSA Fee Considerations
======================

We choose to maintain the native ZEC Asset as the primary token for the Zcash blockchain, similar to how ETH is needed for ERC20 transactions to the benefit of the Ethereum ecosystem.

An alternative proposal for the ZSA fee mechanism that was not adopted was to adopt a new type of fee, denominated in the custom Asset being issued or transferred.

> In the context of transparent transactions, such a fee  allows to “tap into” the value of the transactions for the benefit of the miners. 

Reviewer: There seems to be a missing word here.

Did you mean something like:
```suggestion
In the context of transparent transactions, such a fee allows miners to “tap into” the ZSA value of the transactions, rather than the ZEC value of transactions.
```
